generator/views/view.py

```diff
@@ -104,26 +104,31 @@ def to_lookml(self, bq_client, v1_name: Optional[str]) -> Dict[str, Any]:
 
     def get_client_id(self, dimensions: List[dict], table: str) -> Optional[str]:
         """Return the first field that looks like a client identifier."""
-        client_id_fields = [
-            d["name"]
-            for d in dimensions
-            if d["name"] in {"client_id", "client_info__client_id", "context_id"}
-        ]
-        if not client_id_fields:
-            # Some pings purposely disinclude client_ids, e.g. firefox installer
-            return None
-        if len(client_id_fields) > 1:
-            raise ClickException(f"Duplicate client_id dimension in {table!r}")
-        return client_id_fields[0]
+        client_id_fields = self.select_dimension(
+            {"client_id", "client_info__client_id", "context_id"},
+            dimensions,
+            table,
+        )
+        return client_id_fields["name"] if client_id_fields else None
```

Comment:
I think this comment is more appropriate here:
```suggestion
        # Some pings purposely disinclude client_ids, e.g. firefox installer
        return client_id_fields["name"] if client_id_fields else None
```